arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java

```diff
@@ -309,21 +308,17 @@ public Response geohashgeoaggregate(
                 // if sizes are not equals, it means one multi-value pwithin does not intersects bbox => no results
                 && pwithin.size() == simplifiedPwithin.size()) {
             simplifiedPwithin.add(pwithinBbox);
-            return this.geoaggregate(
-                    collection,
-                    agg,
-                    f,
-                    q,
-                    simplifiedPwithin,
-                    gwithin,
-                    gintersect,
-                    notpwithin,
-                    notgwithin,
-                    notgintersect,
-                    partitionFilter,
-                    pretty,
-                    flat,
-                    maxagecache);
+            AggregationsRequest aggregationsRequest = new AggregationsRequest();
+            aggregationsRequest.filter = ParamsParser.getFilter(f, q, simplifiedPwithin, gwithin, gintersect, notpwithin, notgwithin, notgintersect);
+            aggregationsRequest.aggregations = ParamsParser.getAggregations(agg);
+            AggregationsRequest aggregationsRequestHeader = new AggregationsRequest();
+            aggregationsRequestHeader.filter = ParamsParser.getFilter(partitionFilter);
+            MixedRequest request = new MixedRequest();
+            request.basicRequest = aggregationsRequest;
+            request.headerRequest = aggregationsRequestHeader;
+            FeatureCollection fc = getFeatureCollection(request, collectionReference, (flat != null && flat), Optional.of(geohash));
```

Comment:
You have Boolean.TRUE.equals(null), or with Apache BooleanUtils.isTrue(), in order to avoid testing if a Boolean is null

Reply:
Done